libcloudforensics/providers/gcp/internal/project.py

```diff
@@ -0,0 +1,681 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library for incident response operations on Google Cloud Compute Engine.
+
+Library to make forensic images of Google Compute Engine disk and create
+analysis virtual machine to be used in incident response.
+"""
+
+from __future__ import unicode_literals
+
+import json
+import ssl
+import time
+
+from googleapiclient.errors import HttpError
+
+from libcloudforensics.providers.gcp.internal.common import GenerateDiskName, \
+  log, CreateService, ReadStartupScript
+from libcloudforensics.providers.gcp.internal.compute import \
+  GoogleComputeInstance, GoogleComputeDisk, GoogleComputeImage
+
+
+class GoogleCloudProject:
+  """Class representing a Google Cloud Project.
+
+  Attributes:
+    project_id: Project name.
+    default_zone: Default zone to create new resources in.
+    gce_api_client: Client to interact with GCE APIs.
+
+  Example use:
+    gcp = GoogleCloudProject("your_project_name", "us-east")
+    gcp.ListInstances()
+  """
+
+  COMPUTE_ENGINE_API_VERSION = 'v1'
+
+  def __init__(self, project_id, default_zone=None):
+    """Initialize the GoogleCloudProject object.
+
+    Args:
+      project_id (str): The name of the project.
+      default_zone (str): Optional. Default zone to create new resources in. N
+          one means GlobalZone.
+    """
+
+    self.project_id = project_id
+    self.default_zone = default_zone
+    self.gce_api_client = None
+
+  def GceApi(self):
```

Comment:
should we split this file per API services and move each service to a separate file, similar to log.py, something like gce.py, gcf.py, gcb.py and then the user can optionally initialize an object created by each of these modules inside the project. Later the caller will do something like project.compute.create_instance. This will make porting new functions easy since it is structured in a way similar to the GCP APIs.